File: example/android/awn_core/src/main/java/me/carda/awesome_notifications/core/models/NotificationLocalizationModel.java
package me.carda.awesome_notifications.core.models;

import android.content.Context;

import java.util.HashMap;
import java.util.Map;

import me.carda.awesome_notifications.core.Definitions;
import me.carda.awesome_notifications.core.exceptions.AwesomeNotificationsException;

public class NotificationLocalizationModel extends AbstractModel {
    private static final String TAG = "NotificationLocalization";

    public String title;
    public String body;
    public String summary;
    public String largeIcon;
    public String bigPicture;
    public Map<String, String> buttonLabels;

    @Override
    public NotificationLocalizationModel fromMap(Map<String, Object> arguments) {
        title        = getValueOrDefault(arguments, Definitions.NOTIFICATION_TITLE, String.class, null);
        body         = getValueOrDefault(arguments, Definitions.NOTIFICATION_BODY, String.class, null);
        summary      = getValueOrDefault(arguments, Definitions.NOTIFICATION_SUMMARY, String.class, null);
        largeIcon    = getValueOrDefault(arguments, Definitions.NOTIFICATION_LARGE_ICON, String.class, null);
        bigPicture   = getValueOrDefault(arguments, Definitions.NOTIFICATION_BIG_PICTURE, String.class, null);
        buttonLabels = getValueOrDefaultMap(arguments, Definitions.NOTIFICATION_BUTTON_LABELS, null);

        return this;
    }

    @Override
    public Map<String, Object> toMap() {
        Map<String, Object> dataMap = new HashMap<>();

        putDataOnSerializedMap(Definitions.NOTIFICATION_TITLE, dataMap, title);
        putDataOnSerializedMap(Definitions.NOTIFICATION_BODY, dataMap, body);
        putDataOnSerializedMap(Definitions.NOTIFICATION_SUMMARY, dataMap, summary);
        putDataOnSerializedMap(Definitions.NOTIFICATION_LARGE_ICON, dataMap, largeIcon);
        putDataOnSerializedMap(Definitions.NOTIFICATION_BIG_PICTURE, dataMap, bigPicture);
        putDataOnSerializedMap(Definitions.NOTIFICATION_BUTTON_LABELS, dataMap, buttonLabels);

        return dataMap;
    }

    @Override
    public String toJson() {
        return templateToJson();
    }

    @Override
    public NotificationLocalizationModel fromJson(String json){
        return (NotificationLocalizationModel) super.templateFromJson(json);
    }

    @Override
    public void validate(Context context) throws AwesomeNotificationsException {

    }
}

File: packages/awesome_notifications/ios/AwnCore/Classes/models/NotificationLocalizationModel.swift
//
//  NotificationLocalization.swift
//  IosAwnCore
//
//  Created by Rafael Setragni on 19/02/23.
//

import Foundation

public class NotificationLocalizationModel : AbstractModel {
    
    private static let TAG = "NotificationLocalization"
    
    public var content:NotificationContentModel?
    public var actionButtons:[NotificationButtonModel]?
    public var schedule:NotificationScheduleModel?
    public var importance:NotificationImportance?
    
    var title:String?
    var body:String?
    var summary:String?
    var largeIcon:String?
    var bigPicture:String?
    var buttonLabels:[String:String]?
    
    public init(){}
    
    public func fromMap(arguments: [String : Any?]?) -> AbstractModel? {
        
        self.title      = MapUtils<String>.getValueOrDefault(reference: Definitions.NOTIFICATION_TITLE, arguments: arguments)
        self.body       = MapUtils<String>.getValueOrDefault(reference: Definitions.NOTIFICATION_BODY, arguments: arguments)
        self.summary    = MapUtils<String>.getValueOrDefault(reference: Definitions.NOTIFICATION_SUMMARY, arguments: arguments)
        self.largeIcon  = MapUtils<String>.getValueOrDefault(reference: Definitions.NOTIFICATION_LARGE_ICON, arguments: arguments)
        self.bigPicture = MapUtils<String>.getValueOrDefault(reference: Definitions.NOTIFICATION_BIG_PICTURE, arguments: arguments)
        self.buttonLabels = MapUtils<[String:String]>.getValueOrDefault(reference: Definitions.NOTIFICATION_BUTTON_LABELS, arguments: arguments)
        
        return self
    }
    
    public func toMap() -> [String : Any?] {
        var mapData:[String: Any?] = [:]
        
        if(title != nil) {mapData[Definitions.NOTIFICATION_TITLE] = title}
        if(body != nil) {mapData[Definitions.NOTIFICATION_BODY] = body}
        if(summary != nil) {mapData[Definitions.NOTIFICATION_SUMMARY] = summary}
        if(largeIcon != nil) {mapData[Definitions.NOTIFICATION_LARGE_ICON] = largeIcon}
        if(bigPicture != nil) {mapData[Definitions.NOTIFICATION_BIG_PICTURE] = bigPicture}
        if(buttonLabels != nil) {mapData[Definitions.NOTIFICATION_BUTTON_LABELS] = buttonLabels}
        
        return mapData
    }
    
    public func validate() throws {
    }
}
